### PR TITLE
Add: upload via tus

### PR DESCRIPTION
**Pull request by frosch123**

### CODE REVIEW

File: webclient/pages/version_info.py [hunk 1]
```diff
@@ -296,5 +301,11 @@ def manager_new_package_upload(session, token):
         accept_tos=accept_tos,
         deps_editable=deps_editable,
         messages=messages,
+        tus_url=tus_url(),
+        upload_token=token,
         csrf_token=csrf_token,
     )
+    tus = tus_host()  # None in production
```
Comment (TrueBrain) [inline webclient/pages/version_info.py:308]: This comment can be a lie. You don't know :D And mostly, I don't understand why CSP is disabled if we would be to run in production. Sounds odd? This needs a bit more comment I think.

File: webclient/static/uploader.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    button_validate.disabled = upload_queue.length > 0;
+    button_publish.disabled = upload_queue.length > 0;
+    if (upload_queue.length == 0) return;
+
+    upload_file = upload_queue[0];
+    upload_queue.shift();
+
+    let options = {
+        endpoint: tus_url,
+        retryDelays: [0, 1000, 3000, 5000],
```
Comment (TrueBrain) [inline webclient/static/uploader.js:109]: Is this needed, or are there defaults?

File: webclient/pages/version_info.py (hunk 1 above)
Comment (frosch123) [reply webclient/pages/version_info.py:308]: template() sets a default, which is as strict as it gets.
If tus is not the same as api, then it is slightly relaxed here.

Comment (TrueBrain) [reply webclient/pages/version_info.py:308]: That should go into the comment :D

File: webclient/static/uploader.js (hunk 2 above)
Comment (frosch123) [reply webclient/static/uploader.js:109]: The default is to not retry at all.

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,5 @@
 __pycache__/
 *.pyc
 /.env
+/node_modules
+/package-lock.json
```
Comment (TrueBrain) [inline .gitignore:5]: Please commit this file to the repo too. It is like a `requirements.txt`, and pins the versions you used. This is also the advise npm gives ;)

File: package.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "bananas-frontend-web",
+  "version": "1.0.0",
+  "description": "Bananas frontend",
+  "main": "index.js",
```
Comment (TrueBrain) [inline package.json:5]: as this file does not exist, does this entry need to be here? Or does npm allow it to be missing?

File: Dockerfile [hunk 5]
```diff
@@ -27,6 +37,8 @@ RUN pip freeze 2>/dev/null > requirements.installed \
         && exit 1 ) 1>&2
 
 COPY webclient /code/webclient
+COPY --from=nodejs /code/node_modules/tus-js-client/dist/tus.min.js /code/webclient/static/tus.min.js
+COPY --from=nodejs /code/node_modules/tus-js-client/dist/tus.min.js.map /code/webclient/static/tus.min.js.map
 
 ENTRYPOINT ["python", "-m", "webclient"]
-CMD ["--authentication-method", "developer", "--developer-username", "developer", "--api-url", "http://localhost:8080", "--frontend-url", "http://localhost:5000", "run", "-p", "80", "-h", "0.0.0.0"]
+CMD ["--authentication-method", "github", "--api-url", "https://api.bananas.staging.openttd.org", "--frontend-url", "http://localhost:5000", "run", "-p", "80", "-h", "0.0.0.0"]
```
Comment (frosch123) [inline Dockerfile:44]: Did you intend to change that?
I thought the Dockerfile defaults were for development.

Comment (TrueBrain) [reply Dockerfile:44]: Yeah, this is a bit of a PITA. "localhost" in a docker context is not the "localhost" in our normal context. And there isn't really a hostname that would work .. so there is no way to really connect to a development API this way, without a user entering an IP somewhere.

So I think I have to rethink my "developer" strategy for Dockerfiles. For sure what was there wasn't working .. and at least this is :P

**Pull request merged**